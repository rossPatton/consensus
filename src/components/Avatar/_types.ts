export type tProps = {
  alt?: string,
  className?: string,
  hash?: string,
  size?: string, // number as string
  type?: 'group' | 'user',
};
