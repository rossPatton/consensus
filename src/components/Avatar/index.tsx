import React, {memo} from 'react';

import {tProps} from './_types';

const prefix = 'https://seccdn.libravatar.org/avatar';

// the component half of the search filter container
// decoupled for ease of placement
const Avatar = memo((props: tProps) => {
  if (typeof props.url !== 'string') return null;

  return (
    <img
      alt={props.alt || 'User or Group Avatar'}
      height={props.size || '96'}
      className="bg-white rounded-circ ovrHide"
      src={`${prefix}/${props.url}?default=${props.type === 'group' ? 'identicon' : 'mm'}&size=${props.size}`}
      width={props.size || '96'}
    />
  );
});

export default Avatar;
