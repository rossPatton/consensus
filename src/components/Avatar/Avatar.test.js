import React from 'react';
import render from 'react-test-renderer';

import Avatar from '.';

it('renders without crashing', () => {
  render.create(<Avatar />);
});

it('renders empty div when no props', () => {
  const component = render.create(<Avatar />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders image tag when hash prop provided', () => {
  const component = render.create((
    <Avatar
      hash="notARealHash"
    />
  ));
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
