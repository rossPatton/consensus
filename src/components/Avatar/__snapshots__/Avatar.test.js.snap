// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty div when no props 1`] = `
<div
  className="mr-1 bg-white border p-3 rounded"
/>
`;

exports[`renders image tag when hash prop provided 1`] = `
<div
  className="mr-1"
>
  <img
    alt="User or Group Avatar"
    height="96"
    src="https://seccdn.libravatar.org/avatar/notARealHash?default=mm&size=undefined"
    width="96"
  />
</div>
`;
